proxy/reverse.go

```diff
@@ -203,3 +209,33 @@ func maybeSetForwardedFor(req *http.Request) {
 	}
 	req.Header.Set("X-Forwarded-For", clientIP)
 }
+
+// checkProxyLoop detects a proxy loop by passing around proxyID
+// and endpoints, and call refresh methods to update endpoints
+// if there's any proxy loop found. It returns true if it has
+// found a proxy loop from a endpoint string.
+func (p *reverseProxy) checkProxyLoop(req *http.Request, ep string) bool {
```

Comment:
This name does not match its behavior. The function does check, set and ban. Either you should find an good naming about it, or you should split it into multiple functions.